tsgo --incremental --watch crashes

## Stack trace

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x1011a7fa8]

goroutine 1 [running]:
github.com/microsoft/typescript-go/internal/incremental.(*buildInfoReader).ReadBuildInfo(0x140001ae2c8?, {0x14000461130, 0x41})
        github.com/microsoft/typescript-go/internal/incremental/incremental.go:24 +0x28
github.com/microsoft/typescript-go/internal/incremental.ReadBuildInfoProgram(0x14000168000, {0x1015682a0, 0x14000032050})
        github.com/microsoft/typescript-go/internal/incremental/incremental.go:52 +0x78
github.com/microsoft/typescript-go/internal/execute.(*Watcher).start(0x1400001e360)
        github.com/microsoft/typescript-go/internal/execute/watcher.go:49 +0x170
github.com/microsoft/typescript-go/internal/execute.tscCompilation({0x10156d9d0, 0x1400001e7e0}, 0x140001680f0, 0x0)
        github.com/microsoft/typescript-go/internal/execute/tsc.go:191 +0xa18
github.com/microsoft/typescript-go/internal/execute.CommandLine({0x10156d9d0, 0x1400001e7e0}, {0x1400001e190, 0x5, 0x5}, 0x0)
        github.com/microsoft/typescript-go/internal/execute/tsc.go:63 +0x1c0
main.runMain()
        github.com/microsoft/typescript-go/cmd/tsgo/main.go:23 +0x120
main.main()
        github.com/microsoft/typescript-go/cmd/tsgo/main.go:10 +0x1c
```

## Steps to reproduce

1. clone vscode
2. run `tsgo -p src --noEmit --incremental --watch`

cc @sheetalkamat

--- a/internal/execute/tscwatch_test.go
+++ b/internal/execute/tscwatch_test.go
@@ -15,6 +15,14 @@ func TestWatch(t *testing.T) {
 			},
 			commandLineArgs: []string{"index.ts", "--watch"},
 		},
+		{
+			subScenario: "watch with tsconfig and incremental",
+			files: FileMap{
+				"/home/src/workspaces/project/index.ts":      "",
+				"/home/src/workspaces/project/tsconfig.json": "{}",
+			},
+			commandLineArgs: []string{"--watch", "--incremental"},
+		},
 	}
 
 	for _, test := range testCases {

--- a/internal/execute/watcher.go
+++ b/internal/execute/watcher.go
@@ -42,10 +42,7 @@ func createWatcher(sys System, configParseResult *tsoptions.ParsedCommandLine, r
 }
 
 func (w *Watcher) start() {
-	// if this function is updated, make sure to update `StartForTest` in export_test.go as needed
-	if w.configFileName == "" {
-		w.host = compiler.NewCompilerHost(w.sys.GetCurrentDirectory(), w.sys.FS(), w.sys.DefaultLibraryPath(), nil)
-	}
+	w.host = compiler.NewCompilerHost(w.sys.GetCurrentDirectory(), w.sys.FS(), w.sys.DefaultLibraryPath(), nil)
 	w.program = incremental.ReadBuildInfoProgram(w.options, incremental.NewBuildInfoReader(w.host))
 
 	if !w.testing {

--- a/testdata/baselines/reference/tscWatch/commandLineWatch/watch-with-tsconfig-and-incremental.js
+++ b/testdata/baselines/reference/tscWatch/commandLineWatch/watch-with-tsconfig-and-incremental.js
@@ -1,0 +1,41 @@
+currentDirectory::/home/src/workspaces/project
+useCaseSensitiveFileNames::true
+Input::
+//// [/home/src/workspaces/project/index.ts] *new* 
+
+//// [/home/src/workspaces/project/tsconfig.json] *new* 
+{}
+
+tsgo --watch --incremental
+ExitStatus:: Success
+Output::
+//// [/home/src/tslibs/TS/Lib/lib.d.ts] *Lib*
+/// <reference no-default-lib="true"/>
+interface Boolean {}
+interface Function {}
+interface CallableFunction {}
+interface NewableFunction {}
+interface IArguments {}
+interface Number { toExponential: any; }
+interface Object {}
+interface RegExp {}
+interface String { charAt: any; }
+interface Array<T> { length: number; [n: number]: T; }
+interface ReadonlyArray<T> {}
+interface SymbolConstructor {
+    (desc?: string | number): symbol;
+    for(name: string): symbol;
+    readonly toStringTag: symbol;
+}
+declare var Symbol: SymbolConstructor;
+interface Symbol {
+    readonly [Symbol.toStringTag]: string;
+}
+declare const console: { log(msg: any): void; };
+//// [/home/src/workspaces/project/index.js] *new* 
+
+
+SemanticDiagnostics::
+*refresh*    /home/src/tslibs/TS/Lib/lib.d.ts
+*refresh*    /home/src/workspaces/project/index.ts
+Signatures::

